### PR TITLE
Added Jenkinsfile

### PR DESCRIPTION
**Pull request by payal-sarnot**
Added Jenkinsfile

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                userRemoteConfigs: scm.userRemoteConfigs,
+                branches: scm.branches,
+                doGenerateSubmoduleConfigurations: true,
+                submoduleCfg: scm.submoduleCfg,
+                browser: scm.browser,
+                gitTool: scm.gitTool,
+                extensions: scm.extensions + [
+                    [$class: 'CleanCheckout'],
+                    [$class: 'PruneStaleBranch'],
+                    [$class: "RelativeTargetDirectory", relativeTargetDir: "${WORKSPACE}/myworkspace"],
```
Comment (nikhil-zinjurde-imgtec) [inline Jenkinsfile]: why not name the target directory with the current repo name itself i.e. w1-switchcounter as this jenkinsfile is specific to this repo?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Build Switchcounter Workshop
+ *
+ */
+
+properties([
+    buildDiscarder(logRotator(numToKeepStr: '30')),
+    parameters([
+        stringParam(defaultValue: 'http://downloads.creatordev.io/creator-openwrt/v1/pistachio/marduk/creator-sdk-1.tar.bz2',
```
Comment (Ham22) [inline Jenkinsfile:11]: There is a v2 now right?

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            description: 'Creator image tarball to use', name: "CREATOR_IMAGE"),
+    ])
+])
+
+node('docker && imgtec') {
+    def docker_image
+    docker_image = docker.image "imgtec/workshop-builder:latest"
+    docker_image.inside {
+        stage('Prepare') {
+            checkout([$class: 'GitSCM',
```
Comment (Ham22) [inline Jenkinsfile]: Do you really need all the extra options in this case?

Instead you could replace all of this with just `checkout scm`, also I would suggest download the sdk first as it just seems a bit more logical as that is the manual process followed to build a package with the SDK?

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    buildDiscarder(logRotator(numToKeepStr: '30')),
+    parameters([
+        stringParam(defaultValue: 'http://downloads.creatordev.io/creator-openwrt/v1/pistachio/marduk/creator-sdk-1.tar.bz2',
+            description: 'Creator image tarball to use', name: "CREATOR_IMAGE"),
+    ])
+])
+
+node('docker && imgtec') {
+    def docker_image
+    docker_image = docker.image "imgtec/workshop-builder:latest"
```
Comment (Ham22) [inline Jenkinsfile]: Just use the standard creator docker?

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                browser: scm.browser,
+                gitTool: scm.gitTool,
+                extensions: scm.extensions + [
+                    [$class: 'CleanCheckout'],
+                    [$class: 'PruneStaleBranch'],
+                    [$class: "RelativeTargetDirectory", relativeTargetDir: "${WORKSPACE}/Switch-Counter"],
+                ],
+            ])
+
+            sh "wget -qO- ${params.CREATOR_IMAGE} | tar -xvj --strip-components 1"
```
Comment (Ham22) [inline Jenkinsfile]: Remove -v, pointless, slows it down and add size to the logs

File: Jenkinsfile (hunk 2 above)
Comment (payal-sarnot) [reply Jenkinsfile:11]: The SDK didn't change between creator release v1 and v2, so the squad chose not to put out a fresh version/copy in the creator v2 directory.

Comment (nikhil-zinjurde-imgtec) [reply Jenkinsfile:11]: I believe the SDK v2 did have a major change from v1, https://github.com/CreatorDev/openwrt/commit/e973475ee378c9cf57ded93600ff5c7eb64a9565, so highly recommend to use v2

File: Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            sh "wget -qO- ${params.CREATOR_IMAGE} | tar -xj --strip-components 2"
+            sh "echo 'src-link workshop ../workshop' >> feeds.conf.default"
+            sh "cat Config.in feeds.conf.default"
+            sh "scripts/feeds update -a && scripts/feeds install -a"
+        }
+        stage('Build') {
+          sh "make -j1 V=s package/switch-counter/compile"
+        }
+        stage('Upload') {
+            archiveArtifacts 'bin/pistachio/packages/switch-counter/*'
```
Comment (nikhil-zinjurde-imgtec) [inline Jenkinsfile]: You can change this to '**/switch-counter/*'

**Review by nikhil-zinjurde-imgtec — APPROVED** [review]

**Pull request merged**